src/main/java/org/mitre/synthea/engine/Logic.java

```diff
@@ -73,18 +76,51 @@ public boolean test(Person person, long time) {
   }
   
   /**
-   * The Date condition type tests the current year being simulated. For example, this may be used
-   * to drive different logic depending on the suggested medications or procedures of different time
-   * periods, or model different frequency of conditions.
+   * The Date condition type tests the current year, month, or date being simulated.
+   * For example, this may be used to drive different logic depending on the suggested
+   * medications or procedures of different time periods, or model different frequency
+   * of conditions.
    */
   public static class Date extends Logic {
+    private String datetype;
```

Comment:
Adding a new field to say which thing you're using works just fine, but as you've seen it means that you have to update every existing module to be compliant.   
Another way I would suggest is to instead change the `year` and `month` fields to `Integer` instead of `int`, and then test against whichever one is actually provided, i.e. not null. That gives you the benefit of backward compatibility.